packages/hash/datastore/postgres/migration/1649061269485_add-aggregation-id.ts

```diff
@@ -0,0 +1,59 @@
+import { MigrationBuilder, ColumnDefinitions } from "node-pg-migrate";
+
+export const shorthands: ColumnDefinitions | undefined = undefined;
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  /**
+   * Step 1. add the `aggregation_id` column if it doesn't already exist.
+   */
+  pgm.addColumn(
+    "aggregations",
+    {
+      aggregation_id: {
+        type: "uuid",
+        unique: true,
+      },
+    },
+    {
+      ifNotExists: true,
+    },
+  );
+
+  /**
+   * Step 2. genereate UUID values for each existing aggregation.
+   */
+  pgm.sql(`
+   update aggregations 
+   set aggregation_id = gen_random_uuid()
+   where aggregation_id is null
+ `);
+
+  /**
+   * Step 3. make the `aggregation_id` column non-nullable
+   */
+  pgm.alterColumn("aggregations", "aggregation_id", {
+    notNull: true,
+  });
+
+  /**
+   * Step 4. replace existing primary key for the `aggregations` table
+   */
+  pgm.dropConstraint("aggregations", "aggregations_pkey", { ifExists: true });
+  pgm.addConstraint("aggregations", "aggregations_pkey", {
+    primaryKey: "aggregation_id",
+  });
```

Comment:
Don't we still need `source_account_id` for sharding?

Reply:
Good catch, yes my understanding based on prior discussions is that the sharding key has to be in the primary key for a distributed table. Have fixed this in https://github.com/hashintel/hash/pull/465/commits/4ef7172bc6d8b48fa0a3416a6b0239aebcdaeae6